### PR TITLE
CARRY: Add Kueue webhook NetworkPolicy

### PR DESCRIPTION
**Pull request by astefanutti**
Fixes https://issues.redhat.com/browse/RHOAIENG-3415.

Depends on #18.

### CODE REVIEW

**Comment by zdtsw** [comment]
> /cc @zdtsw

meaning:  any incoming traffic can reach any pods in namespace (upstream opendatahub, downstream redhat-ods-applications) on port 9443? 
does it need to be so wildly opened?

**Comment by astefanutti** [comment]
> > /cc @zdtsw
> 
> meaning: any incoming traffic can reach any pods in namespace (upstream opendatahub, downstream redhat-ods-applications) on port 9443? does it need to be so wildly opened?

Kustomize applies common labels so the rendered NetworkPolicy only open traffic for Kueue Pods:

```
apiVersion: networking.k8s.io/v1
kind: NetworkPolicy
metadata:
  labels:
    app.kubernetes.io/component: controller
    app.kubernetes.io/name: kueue
  name: kueue-kueue-webhook-server
  namespace: opendatahub
spec:
  ingress:
  - ports:
    - port: 9443
      protocol: TCP
  podSelector:
    matchLabels:
      app.kubernetes.io/component: controller
      app.kubernetes.io/name: kueue
  policyTypes:
  - Ingress
 ```

File: config/rhoai/webhook_network_policy.yaml [hunk 1]
```diff
@@ -0,0 +1,14 @@
+kind: NetworkPolicy
+apiVersion: networking.k8s.io/v1
+metadata:
+  name: kueue-webhook-server
```
Comment (sutaakar) [inline config/rhoai/webhook_network_policy.yaml:4]:
```suggestion
  name: webhook-server
```
Kustomize attach `kueue` prefix

Comment (astefanutti) [reply config/rhoai/webhook_network_policy.yaml]: Good catch, thanks! Updated.

**Review by sutaakar — APPROVED** [review]

**Pull request merged**